BEAMS3D/Sources/beams3d_physics_mod.f90

```diff
@@ -694,15 +694,15 @@ SUBROUTINE beams3d_follow_neut(t, q)
          LOGICAL          :: ltest
          INTEGER          :: ier, l, m,o
          DOUBLE PRECISION :: rinv, phi_temp, dt_local, ti_temp, ne_temp,&
-                             s_temp, x0, y0, z0, xw, yw, zw, te_temp, Zeff_temp,&
-							 rlim, zlim
-         DOUBLE PRECISION :: qf(3),qs(3),qe(3)
-         DOUBLE PRECISION :: rlocal(num_depo), plocal(num_depo), zlocal(num_depo)
-         DOUBLE PRECISION :: tilocal(num_depo), telocal(num_depo), nelocal(num_depo)
-         DOUBLE PRECISION :: zefflocal(num_depo)
-         DOUBLE PRECISION :: nilocal(NION,num_depo)
-         DOUBLE PRECISION :: tau_inv(num_depo), energy(num_depo)
-         DOUBLE PRECISION :: sigvii(num_depo), sigvcx(num_depo), sigvei(num_depo)
+                           s_temp, x0, y0, z0, xw, yw, zw, te_temp, Zeff_temp,&
+							      rlim, zlim
+         DOUBLE PRECISION ::qf(3),qs(3),qe(3)
+         DOUBLE PRECISION, DIMENSION(:), ALLOCATABLE :: rlocal, plocal, zlocal,&
```

Comment:
Please revert these.  The size is always known and there's no reason to allocate/deallocate them.